lisa/sut_orchestrator/libvirt/features.py

```diff
@@ -55,3 +55,9 @@ def on_before_deployment(cls, *args: Any, **kwargs: Any) -> None:
             node_context.guest_vm_type = cls._security_profile_mapping[
                 settings.security_profile
             ]
+
+
+class CVMNestedVirtualization(features.CVMNestedVirtualization):
+    @classmethod
+    def create_setting(cls, *args: Any, **kwargs: Any) -> schema.FeatureSettings:
+        return schema.FeatureSettings.create(cls.name())
```

Comment:
Instead of returning the settings right away, can we have a check on `host_node` if it supports nested cvm ?

Reply:
> can we have a check on host_node if it supports nested cvm ?

Is that something a feature is supposed to do? I would imagine the code that adds this feature to a node should do those checks (if at all).

Reply:
From this ref: https://github.com/microsoft/lisa/blob/main/lisa/sut_orchestrator/azure/features.py#L2746, I understand the return from this function would say if the actual feature (here in this case Nested CVM) is supported or not. Not all the `host_node` s support nested CVM.
We could also do it as you suggested, but the checks to support feature should be necessary.

Reply:
Ref:https://github.com/microsoft/lisa/blob/ceac54df113ecc1b29f6af83d521171f19ae7b9d/lisa/sut_orchestrator/libvirt/platform.py#L353

Returning ```None``` for the case when CVM is not supported, we will have to set the return type as Optional[schema.FeatureSettings].  Whereas ```node_capabilities.features``` expects value of type ```schema.FeatureSettings```.

Reply:
> For returning `None` for the case when CVM is not supported, we will have to set the return type as Optional[schema.FeatureSettings]. Whereas `node_capabilities.features` expects value of type `schema.FeatureSettings`.

You can see how Azure platform handles this and do something similar. I see Azure platform also returning `Optional[schema.FeatureSettings]`.
ref: https://github.com/microsoft/lisa/blob/main/lisa/sut_orchestrator/azure/features.py#L2740

I'm fine with Anirudh's suggestion to have the check before adding the feature to supported list.

>I believe we have a testcase- verify_azure_vm_snp_enablement as part of host functional test job, can't we use the existing testcase to validate?

Testcase and Features are different entities altogether. We cannot use testcase as check if a feature is supported or not.